model/connect/Database.php

```diff
@@ -174,13 +175,29 @@ protected function previewWrite($sql) {
 	 *
 	 * @param string $sql Query to run, and single parameter to callback
 	 * @param callable $callback Callback to execute code
+	 * @param array $params Parameters for any parameterised query
 	 * @return mixed Result of query
 	 */
-	protected function benchmarkQuery($sql, $callback) {
+	protected function benchmarkQuery($sql, $callback, $params = array()) {
 		if (isset($_REQUEST['showqueries']) && Director::isDev()) {
 			$starttime = microtime(true);
 			$result = $callback($sql);
 			$endtime = round(microtime(true) - $starttime, 4);
+			// replace parameters as closely as possible to what we'd expect the DB to put in
+			if (strtolower($_REQUEST['showqueries']) == 'inline') {
```

Comment:
There's already code for this in https://github.com/silverstripe/silverstripe-framework/blob/3.2/model/queries/SQLQuery.php#L162, to save you having to double-implement this. :)

Your code also doesn't handle literal question marks in queries. E.g.

sql: `WHERE "Title" in ('?', ?)`
params: 'MyTitle'